doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_recovery.rst

```diff
@@ -0,0 +1,227 @@
+.. _ug_nrf54h20_suit_recovery:
+
+Configuring and programming the SUIT manufacturer application recovery firmware
+###############################################################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+During the device lifetime, the device may encounter various issues that can lead to boot failure.
+In such cases the device needs a way to recover from the failure and continue to operate.
+
+In SUIT, this is solved by running a specially prepared recovery firmware.
+Although the recovery firmware needs some additional space in the device, it is a highly recommended feature for all devices that use SUIT.
+
+This document describes scenarios in which a nRF54H20 device using SUIT can enter recovery mode and how the recovering process works.
+
+When is recovery mode entered
+******************************
+
+When booting the device, the Secure Domain verifies the currently installed manifests and firmware images.
+If the verification fails, the device enters recovery mode.
+The reasons for validation failure can be (among others):
+
+* Invalid manifest signature.
+* The digest of the installed firmware image does not match the digest in the manifest.
+* The manifest attempts to perform unauthorized operations.
+
+The reasons for the above failures can be:
+
+* Tampering by an attacker.
+* Bitflips in the MRAM memory due to radiation or other external conditions.
+* Hardware failures in advanced cases where an inplace update is performed and a firmware image is partially overwritten.
+* An incorrectly constructed manifest, leading to a successful update but failure during boot.
+
+The last case should never happen in a production environment, but it is possible during development.
+The manufacturer should always ensure that the manifest is correctly constructed and all the images managed by SUIT are compatible with each other after the update.
+Still, some bugs can be missed during the development phase, which could lead to inconsitencies in the firmware.
```

Comment:
```suggestion
.. caution::
  The last case must never occur in a production environment, but it is possible during development.
  The manufacturer must always ensure that the manifest is correctly constructed and that all images managed by SUIT are compatible with each other after the update.
  If any bugs are overlooked during the development phase, this could result in inconsistencies in the firmware.
```